File: backend/models/session.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Session struct{
	ID uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	Name *string `json:"name"` 
	Questions []Question `gorm:"many2many:session_questions;" json:"questions"`
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: frontend/app/types/session.ts
import type { QuestionResponseDto } from "./question";

export interface CreateSessionDto {
  name?: string;
  question_number?: number;
}

export interface SessionResponseDto {
  id: string;
  name: string;
  questions: QuestionResponseDto[];
  created_at: string;
  updated_at: string;
}
